src/lib.rs

```diff
@@ -799,23 +799,18 @@ impl MPCParameters {
                         for (base, projective) in bases.iter_mut().zip(projective.iter_mut()) {
                             *projective = wnaf.base(base.into_projective(), 1).scalar(coeff);
                         }
+
```

Comment:
I may have observed a slight improvement, or it might have been noise. It didn't get worse.

I would be surprised if the other way were better. The first part was assigning the work to specific thread, one per available cpu. So they should all be running in parallel.  Then the work is being chunked back up into (presumably) the same number of threads and continued. It seems to me that this incurs at least:

- the penalty of waiting for all threads to complete before beginning any normalization/conversion.
- the rayon overhead
- possibly moving data between cpus, if work isn't reassigned to the same cpu it started on (despite the same chunking)

As far as 'stealing cpu' goes, the normalization and conversion are single-threaded, so they shouldn't be stealing CPU from other threads — just using the already-assigned idle CPU.

Doing all the work for each chunk on the originally allocated thread at least avoids all the potential costs above. As an aside, if we weren't using the raw format to speed up i/o, we could probably improve performance by moving the affine<->uncompressed conversions into this parallel section. And we could likely get even more speedup than raw by just reading/writing the projective points and skipping the conversions (at the cost of larger small params files). The latter might be worth exploring for future ceremonies.